networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/provider_nsx_policy.py

```diff
@@ -635,13 +635,27 @@ def _setup_default_infrastructure_rules(self):
             path = API.POLICY.format(policy["id"])
             res = self.client.get(path=path)
             if res.ok:
-                continue
+                if self._check_for_infrastructure_changes():
+                    self.client.put(path=path, data=policy).raise_for_status()
             elif res.status_code == 404:
                 LOG.info("Infrastructure Policy %s not found, creating...", policy["display_name"])
                 self.client.put(path=path, data=policy).raise_for_status()
             else:
                 res.raise_for_status()
 
+    def _check_infrastructure_rules_for_updates(self, policies_from_cfg, security_policies):
+        for rule_from_cfg in policies_from_cfg['rules']:
+            for rule_policy in sg_policy["rules"]:
+                if rule_policy['id'] == rule_from_cfg['id']:
+                    shared_keys = set(rule.keys()).intersection(set(rule_policy.keys()))
```

Comment:
[optional] I like using operators for set operations, but it depends of what you like yourself. For intersection you have the option of using `&`, so `set(rule) & set(rule_policy)` (the `.keys()` is implicit for dict-to-list-conversion but that's also a choice on how verbose you like to be, makes it more ovious that you're dealing with a dict).